backend/src/personnel/personnel.service.ts

```diff
@@ -185,20 +185,20 @@ export class PersonnelService {
     const person = await this.personnelRepository.findOne({
       where: { email: req.idir },
       relations: {
-        bcws: preferences.bcws.roles
+        bcws: preferences.bcws?.roles
           ? {
               roles: true,
             }
           : false,
-        emcr: preferences.emcr.experiences
+        emcr: preferences.emcr?.experiences
           ? {
               experiences: true,
             }
           : false,
       },
     });
 
-    if (preferences.bcws) {
+    if (preferences?.bcws) {
```

Comment:
It's fine, but preference seems to be non nullable as the function's parameter so this shouldn't be necessary

Reply:
You're right - I got a little carried away, lol